cognee/infrastructure/databases/vector/pgvector/PGVectorAdapter.py

```diff
@@ -0,0 +1,243 @@
+from typing import List, Optional, get_type_hints
+from sqlalchemy import JSON, Column, Table, select, delete
+
+from ..models.ScoredResult import ScoredResult
+import asyncio
+
+from ..vector_db_interface import VectorDBInterface, DataPoint
+from sqlalchemy.orm import Mapped, mapped_column
+from ..embeddings.EmbeddingEngine import EmbeddingEngine
+from sqlalchemy.ext.asyncio import create_async_engine, async_sessionmaker
+from pgvector.sqlalchemy import Vector
+
+from ...relational.sqlalchemy.SqlAlchemyAdapter import SQLAlchemyAdapter
+from ...relational.ModelBase import Base
+
+from datetime import datetime
+
+
+# TODO: Find better location for function
+def serialize_datetime(data):
```

Comment:
Let's move this function outside in a separate file.

Reply:
Done